cmd/osmosisd/cmd/genesis.go

```diff
@@ -0,0 +1,152 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	claimtypes "github.com/c-osmosis/osmosis/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/spf13/cobra"
+	appparams "github.com/c-osmosis/osmosis/app/params"
+)
+
+func GenerateGenesisCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "export-genesis [input-snapshot-file]",
+		Short: "Export a genesis from fairdrop snapshot",
+		Long: `Export a genesis from fairdrop snapshot
+Example:
+	osmosisd export-genesis ../snapshot.json
+	- Check input genesis:
+		file is at ~/.gaiad/config/genesis.json
+`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			depCdc := clientCtx.JSONMarshaler
+			cdc := depCdc.(codec.Marshaler)
+			aminoCodec := clientCtx.LegacyAmino.Amino
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			snapshotInput := args[0]
+			osdenom := "uosmo"
+
+			genFile := config.GenesisFile()
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			authGenState := authtypes.GetGenesisStateFromAppState(cdc, appState)
+
+			accs, err := authtypes.UnpackAccounts(authGenState.Accounts)
+			if err != nil {
+				return fmt.Errorf("failed to get accounts from any: %w", err)
+			}
+
+			// Read snapshot file
+			snapshotJson, err := os.Open(snapshotInput)
+			if err != nil {
+				return err
+			}
+			defer snapshotJson.Close()
+
+			byteValue, _ := ioutil.ReadAll(snapshotJson)
+
+			// Produce the map of address to total atom balance, both staked and unstaked
+			snapshot := make(map[string]SnapshotFields)
+			err = aminoCodec.UnmarshalJSON(byteValue, &snapshot)
+			if err != nil {
+				return err
+			}
+
+			balances := []banktypes.Balance{}
+			feeBalances := []banktypes.Balance{}
+
+			totalOsmoBalance := sdk.NewInt(0)
+			for _, acc := range snapshot {
+				// calculate total osmo balance
+				totalOsmoBalance = totalOsmoBalance.Add(acc.OsmoBalance)
+
+				// set atom bech32 prefixes
+				setCosmosBech32Prefixes()
+				
+				// read address from snapshot
+				address, err := sdk.AccAddressFromBech32(acc.AtomAddress)
+				if err != nil {
+					return err
+				}
+ 
+				// set osmo bech32 prefixes
+				appparams.SetBech32Prefixes()
+
+				// airdrop balances
+				coins := sdk.NewCoins(sdk.NewCoin(osdenom, acc.OsmoNormalizedBalance))
```

Comment:
Why is this using NormalizedBalance why `totalOsmoBalance` (on line 83) is using `OsmoBalance`?

Reply:
It was an issue, resolved to totalNormalizedOsmoBalance which sum up OsmoNormalizedBalance not OsmoBalance - confusion for variable names.